### PR TITLE
Add classinfo query to schema endpoint

### LINKED ISSUE #75: Add schema classinfo parameter to app schema endpoint
The schema endpoint should take a ``classinfo`` param such that any class's config can be retrieved. Including ``Input`` & ``Output`` schemas.

### PR DESCRIPTION
**Pull request by jamienoss**
Resolves #75 

Depends on #73

### CODE REVIEW

**Comment by jamienoss** [comment]
@amitschang FYI this is only a draft as it's dependent on, and against, #73 but is otherwise ready to review, please (just not merge), cheers.

**Comment by jamienoss** [comment]
@amitschang This is good for review now please, cheers.

File: evolver/app/main.py [hunk 1]
```diff
@@ -53,9 +56,28 @@ async def update_evolver(config: EvolverConfigWithoutDefaults):
     app.state.evolver.config_model.save(app_settings.CONFIG_FILE)
 
 
-@app.get("/schema")
-async def get_schema():
-    return app.state.evolver.schema
+class SchemaResponse(pydantic.BaseModel):
```
Comment (amitschang) [inline evolver/app/main.py:59]: Could be a good time to create an `evolver/app/model.py` module with this in there

File: evolver/app/tests/test_app.py [hunk 2]
```diff
@@ -61,11 +65,31 @@ def test_evolver_app_control_loop_setup(self, app_client):
...
+            NoOpSensorDriver,
+        ),
+    )
+    def test_schema_endpoint(self, app_client, classinfo):
+        fqn = evolver.util.fully_qualified_name(classinfo) if classinfo else evolver.util.fully_qualified_name(Evolver)
+        response = app_client.get("/schema", params=dict(classinfo=fqn) if classinfo else None)
         assert response.status_code == 200
         # There's not much in the default config yet, this will change in future PRs.
-        assert json.loads(response.content) == dict(hardware=[], controllers=[])
+        assert json.loads(response.content) == SchemaResponse(classinfo=fqn).model_dump(mode="json")
```
Comment (amitschang) [inline evolver/app/tests/test_app.py:87]: might be a good anchor to add a test to pass a simple case through `SchemaResponse` and expect the actual output (or at least significant parts of it). This test would succeed even if SchemaResponse returned `{}` always.

**Review by amitschang — COMMENTED** [review]
some comments

File: evolver/app/main.py [hunk 3]
```diff
@@ -57,9 +59,28 @@ async def update_evolver(config: EvolverConfigWithoutDefaults):
...
+        if issubclass(self.classinfo, BaseConfig):
+            self.config = self.classinfo.model_json_schema()
+        elif issubclass(self.classinfo, BaseInterface):
+            self.config = self.classinfo.Config.model_json_schema()
+
+            if hasattr(self.classinfo, "Input") and issubclass(self.classinfo.Input, pydantic.BaseModel):
+                self.input = self.classinfo.Input.model_json_schema()
+
+            if hasattr(self.classinfo, "Output") and issubclass(self.classinfo.Output, pydantic.BaseModel):
+                self.input = self.classinfo.Output.model_json_schema()
```
Comment (jamienoss) [inline evolver/app/main.py:78]: Should be ``self.output = ...``.

**Review by amitschang — APPROVED** [review]
cool, thanks!

**Pull request merged**